cadccutout/cadccutout/core.py

```diff
@@ -236,3 +236,105 @@ def cutout_from_string(self, cutout_dimensions_str, input_reader=sys.stdin,
     def _get_file_helper(self, file_type, input_reader, output_writer):
         return self.helper_factory.get_instance(file_type, input_reader,
                                                 output_writer)
+
+
+class WriteOnlyStream(BufferedRandom):
+    def __init__(self, raw):
+        super(WriteOnlyStream, self).__init__(io.BytesIO())
+        self._raw = raw
+        self.write_offset = 0
+        self.read_offset = 0
+
+    def read(self, size=1):
+        raise ValueError('Unreadable stream.  This is write only.')
+
+    def write(self, data):
+        written = self._raw.write(data)
+        if written:
+            self.write_offset += written
+        self._raw.flush()
+        return self.write_offset
+
+    def tell(self):
+        return self.write_offset
+
+    def seek(self, offset):
+        raise ValueError('Unseekable stream.  This is write only.')
+
+
+def main_app(argv=None):
+    # Execute only if run as a script.
+    parser = argparse.ArgumentParser()
+
+    # Python 3 uses the buffer property to treat stream data as binary.
+    # Python 2 (sometimes) requires the -u command line switch.
+    if hasattr(sys.stdin, 'buffer'):
+        default_input = sys.stdin.buffer
+    else:
+        default_input = sys.stdin
+
+    if hasattr(sys.stdout, 'buffer'):
+        default_output = sys.stdout.buffer
+    else:
+        default_output = sys.stdout
+
+    parser.description = ('Cutout library to extract an N-Dimension array.')
+    parser.formatter_class = argparse.RawTextHelpFormatter
+
+    parser.add_argument('-d', '--debug', action='store_true',
+                        help='debug messages')
+    parser.add_argument('-q', '--quiet', action='store_true',
+                        help='run quietly')
+    parser.add_argument('-v', '--verbose', action='store_true',
+                        help='verbose messages')
+    parser.add_argument('--version', action='version', version='0.1.1')
```

Comment:
You get the version from setup.cfg like this:
from cadccutout import version
version.version

or equivalent.